networkx/algorithms/community/louvain.py

```diff
@@ -204,3 +207,16 @@ def _gen_graph(G, partition):
         temp = H.get_edge_data(com1, com2, {"weight": 0})["weight"]
         H.add_edge(com1, com2, **{"weight": wt + temp})
     return H
+
+
+def _convert_multigraph(G, weight):
+    """Convert a Multigraph to normal Graph"""
+    H = nx.Graph()
+    H.add_nodes_from(G)
+    for u, v, data in G.edges(data=True):
+        wt = data.get(weight, 1)
```

Comment:
The `G.edges` method can do the data lookup. Not sure that it is any faster, but it's a cleaner interface, maybe:
```
    for u, v, wt in G.edges(data=weight, default=1):
        if H.has_edge(u, v):
            H[u][v]["weight"] += wt
        else:
            H.add_edge(u, v, weight=wt)
    return H
```

Reply:
Good catch thanks!